contracts/eosio.bios/src/eosio.bios.cpp

```diff
@@ -20,6 +20,21 @@ void bios::setabi( name account, const std::vector<char>& abi ) {
    }
 }
 
+// helpers for defining std::unordered_set<eosio::bls_g1>
+struct g1_hash {
+   // bls_g1 is defined as std::array<char, 96>
+   std::size_t operator()(const eosio::bls_g1& g1) const {
+      std::hash<std::string> hash_func;
+      return hash_func(g1.data());
+   }
+};
+
+struct g1_equal {
+   bool operator()(const eosio::bls_g1& lhs, const eosio::bls_g1& rhs) const {
+      return std::memcmp(lhs.data(), rhs.data(), 96 * sizeof(char)) == 0;
```

Comment:
Use `lhs.size()` would be better here. Even better to add a `static_assert(lhs.size() == sizeof(bls_g1))`

Reply:
Thanks. Changed to use `lhs.size()`. But the `static_assert` is not possible as lhs.size() is not a constant integral value. The compiler complains
`error: static_assert expression is not an integral constant expression`

Reply:
I moved `g1_hash` and `g1_equal` inside the function body for better encapsulation.